inc/bmp/BMP.h

```diff
@@ -0,0 +1,289 @@
+#ifndef SENSORS_H
+#define SENSORS_H
+
+
+#include "BMP_def.h"
+
+
+
+
+typedef BMP5_INTF_RET_TYPE (*bmp5_read_fptr_t)(uint8_t reg_addr, uint8_t *read_data, uint32_t len, void *intf_ptr);
+typedef BMP5_INTF_RET_TYPE (*bmp5_write_fptr_t)(uint8_t reg_addr, const uint8_t *read_data, uint32_t len,
+                                                void *intf_ptr);
+typedef void (*bmp5_delay_us_fptr_t)(uint32_t period, void *intf_ptr);
+
+enum bmp5_intf {
+    /*! SPI interface */
+    BMP5_SPI_INTF,
+    /*! I2C interface */
+    BMP5_I2C_INTF,
+    /*! I3C interface */
+    BMP5_I3C_INTF
+};
+
+struct BMP581_InterfaceData
+{
+    // Communication interface (I2C or SPI)
+    bmp5_intf interface;
+
+    // I2C settings
+    uint8_t i2cAddress;
+    ////TwoWire* i2cPort;
+    WIRE_TEENSY* i2cPort;
```

Comment:
Changer à I2C_INTERFACE

```suggestion
    I2C_INTERFACE* i2cPort;
```